src/plugins/intel_gpu/src/plugin/transformations/fc_horizontal_fusion.cpp

```diff
@@ -0,0 +1,206 @@
+            for (size_t i = 2; i < num_inputs; ++i) {
+                const auto& fc_input = fc_user->get_input_node_shared_ptr(i);
+                if (!is_constant(fc_input) && !is_placeholder(fc_input))
+                    return false;
+                if (i == 2 && !is_placeholder(fc_input)) {
+                    nodes_with_bias++;
+                }
+            }
+            user_fc_count++;
+        }
+        return (user_fc_count == num_fcs_to_fuse) && (nodes_with_bias == num_fcs_to_fuse || nodes_with_bias == 0) &&
+               (nodes_with_zp == num_fcs_to_fuse || nodes_with_zp == 0);
+    };
+
+    auto target_fc = wrap_type<op::FullyConnectedCompressed>(is_target_pattern);
+
+    ov::matcher_pass_callback callback = [=](Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto m_fc = pattern_map.at(target_fc).get_node_shared_ptr();
+        auto input_node = m_fc->get_input_node_shared_ptr(0);
+        std::vector<std::shared_ptr<op::FullyConnectedCompressed>> fc_nodes;
+        ov::NodeVector weight_nodes;
+        ov::NodeVector scale_nodes;
+        ov::NodeVector bias_nodes;
+        ov::NodeVector zp_nodes;
+        for (auto user : input_node->get_users()) {
+            auto fc_user = std::dynamic_pointer_cast<op::FullyConnectedCompressed>(user);
+            if (fc_user) {
+                OPENVINO_ASSERT(fc_user->inputs().size() >= 4, "Compressed FC should have at least 4 inputs");
+                fc_nodes.push_back(fc_user);
+                weight_nodes.push_back(fc_user->get_input_node_shared_ptr(1));
+                if (!std::dynamic_pointer_cast<op::Placeholder>(fc_user->get_input_node_shared_ptr(2)))
+                    bias_nodes.push_back(fc_user->get_input_node_shared_ptr(2));
+                scale_nodes.push_back(fc_user->get_input_node_shared_ptr(3));
+                if (fc_user->inputs().size() > 4)
+                    zp_nodes.push_back(fc_user->get_input_node_shared_ptr(4));
+            }
+        }
+        auto weight_dtype = fc_nodes[0]->get_input_element_type(1);
+        auto k_size = fc_nodes[0]->get_input_shape(1)[fc_nodes[0]->get_input_shape(1).size() - 1];
+        std::vector<int64_t> orig_n_sizes;
+        // merge weights, scale, zp
+        for (auto fc : fc_nodes) {
+            if (k_size != fc->get_input_shape(1)[fc->get_input_shape(1).size() - 1])
+                return false;
+            if (weight_dtype != fc->get_input_element_type(1))
+                return false;
+            orig_n_sizes.push_back(fc->get_input_shape(1)[fc->get_input_shape(1).size() - 2]);
+        }
+        auto weight_nodes_as_output_vector = ov::OutputVector{weight_nodes[0], weight_nodes[1], weight_nodes[2]};
+        auto fused_weight = std::make_shared<ov::op::v0::Concat>(weight_nodes_as_output_vector, 0);
+        fused_weight->set_friendly_name(weight_nodes[0]->get_friendly_name() + "_fused");
+        ov::copy_runtime_info(weight_nodes[0], fused_weight);
```

Comment:
AFAIK, It doesn't seem to need to merge runtime info unless the node is being removed or replaced

Reply:
Added beacuse new nodes needs runtime info..